apps/app/src/hooks/useCloseOnBreakpoint.ts

```diff
@@ -0,0 +1,28 @@
+import { Dispatch, SetStateAction, useCallback, useEffect, useState } from "react";
+
+type UseCloseOnBreakpointProps = {
+	initialState: boolean;
+	breakpoint: number;
+};
+
+export const useCloseOnBreakpoint = ({
+	initialState,
+	breakpoint,
+}: UseCloseOnBreakpointProps): [boolean, Dispatch<SetStateAction<boolean>>] => {
+	const [isOpen, setIsOpen] = useState(initialState);
+
+	const handleResize = useCallback(() => {
+		if (window.innerWidth > breakpoint) {
+			setIsOpen(false);
+		}
```

Comment:
A jak powiększę ekran i potem go zmniejszę to już się nie włączony z powrotem?

Reply:
Powinniśmy mieć dwa stany: jeden mówiący o tym czy ktoś ustawił wartość na true czy na false oraz drugi mówiący o tym czy breakpoint jest spełniony. Hook powinien zwracać koniunkcję obu tych wartości.